src/fetch/errors.rs

```diff
@@ -8,10 +10,15 @@ pub enum FetchError {
     FailedToWrapHttpResponse(String),
     InvalidRequestBody(String),
     InvalidRequestHeader(String),
-    InvalidResponseFromJS,
-    NoWindow,
+    InvalidResponseFromJs(JsValue),
     RequestFailed(String),
     RequestTimeout(async_std::future::TimeoutError),
     UnableToCreateRequest(String),
     UnableToSetRequestHeader(String),
 }
+
+impl From<JsValue> for FetchError {
```

Comment:
Do you still need this? Seems like `map_err(InvalidResponseFromJs)` is doing the same thing?

Reply:
There are some cases of `?` in the code. Maybe it is better to be more explicit?

Also, maybe `InvalidResponseFromJs` was only intended for `Response`?

Reply:
Ended up removing it and adding more specific errors